doc/hld_template.md

```diff
@@ -0,0 +1,84 @@
+This section should explain how the new feature/enhancement (module/sub-module) fits in the existing architecture. 
+
+### High-Level Design 
+
+This section covers the high level design of the feature/enhancement. This section covers the following points in detail.
+		
+	- What are the modules and sub-modules that are modified for this design?
+	- What are the repositories that would be changed?
+	- Module/sub-module interfaces and dependencies. 
+	- SWSS and Syncd changes in detail
+	- DB and Schema changes (APP_DB, ASIC_DB, COUNTERS_DB, LOGLEVEL_DB, CONFIG_DB, STATE_DB)
+	- Sequence diagram if required.
+	- Linux dependencies and interface
+	- Warm reboot requirements/dependencies
+	- Fastboot requirements/dependencies
+	- Scalability and performance requirements/impact
+	- Memory requirements
+	- Docker dependency
+	- Build dependency if any
+	- Management interfaces - SNMP, CLI, RestAPI, etc.,
+	- Serviceability and Debug (logging, counters, trace etc) related design
+	- Is this change specific to any platform? Are there dependencies for platforms to implement anything to make this feature work? If yes, explain in detail and inform community in advance.
+	- SAI API requirements, CLI requirements, ConfigDB requirements. Design is covered in following sections.
+
+### SAI API 
+
+This section covers the changes made or new API added in SAI API for implementing this feature. If there is no change in SAI API for HLD feature, it should be explicitly mentioned in this section.
+This section should list the SAI APIs/objects used by the design so that silicon vendors can implement the required support in their SAI. Note that the SAI requirements should be discussed with SAI community during the design phase and ensure the required SAI support is implemented along with the feature/enhancement.
+
+### Configuration and management 
+This section should have sub-sections for all types of configuration and management related design. Example sub-sections for "CLI" and "Config DB" are given below. Sub-sections related to data models (YANG, REST, gNMI, etc.,) should be added as required.
+
+#### CLI Enhancements 
+
+This sub-section covers the addition/deletion/modification of CLI changes needed for the feature in detail. If there is no change in CLI for HLD feature, it should be explicitly mentioned in this section. Note that the CLI changes should ensure downward compatibility with the previous/existing CLI. i.e. Users should be able to save and restore the CLI from previous release even after the new CLI is implemented. 
+This should also explain the CLICK and/or KLISH related configuration/show in detail.
+
+#### Config DB Enhancements  
+
+This sub-section covers the addition/deletion/modification of config DB changes needed for the feature. If there is no change in configuration for HLD feature, it should be explicitly mentioned in this section. This section should also ensure the downward compatibility for the change. 
+		
```

Comment:
Hi @kannankvs 
I'm sorry to comment so late but I feel it's better to move the `Config DB enhancements` to a place ahead of or in the section `High-Level Design`. Because it's can help the readers of the HLD understand it better if there is a database description ahead of the flows which are described in the HLD.
How do you think?
Thanks,
Stephen